### PR TITLE
workaround for default profile issue

### PR DESCRIPTION
**Pull request by deepu105**
Ok so I tried many ways to set a default profile in a clean way and seems like there might be a spring boot bug which is causing an issue see https://github.com/spring-projects/spring-boot/issues/5998

So for the time being I have done a workaround using a Util file, its a bit similar to what @erikkemperman was doing earlier

@jdubois are you ok with this? If spring fixes the above mentioned issue then we can remove the Util file.

Ill wait for spring guys to evaluate the issue and if its a bug ill see if I can a do a PR to fix it.

fix https://github.com/jhipster/generator-jhipster/issues/3618

### CODE REVIEW

**Comment by erikkemperman** [comment]
@deepu105 ... And we're back to a helper class. This whole thing is like deja vu all over again :-)

Well, at least there's no extra hidden file. I'm very curious how the guys at Spring-Boot will classify your issue, but in the mean time I agree this is probably better than the current behaviour of master.

**Comment by deepu105** [comment]
@jdubois could you plz merge if you are ok with this. This should fix https://github.com/jhipster/generator-jhipster/issues/3618 as well

**Comment by jdubois** [comment]
Let me fist get back home from Spring IO :-)

**Comment by deepu105** [comment]
Oh ok, hope you had a great time :)

Thanks & Regards,
Deepu

**Comment by deepu105** [comment]
Spring boot guys acknowledged this as a bug so may be we can merge this as a temporaray solution and can remove it when the issue is fixed in spring boot

File: generators/generator-base.js [hunk 1]
```diff
@@ -1252,7 +1252,7 @@ Generator.prototype.printJHipsterLogo = function () {
     );
     this.log(chalk.white.bold('                            http://jhipster.github.io\n'));
     if (this.checkInstall) this.checkForNewVersion();
-    this.log(chalk.white('Welcome to the JHipster Generator ') + chalk.yellow('v' + packagejs.version));
+    this.log(chalk.white('Welcome to the JHipster Generator test') + chalk.yellow('v' + packagejs.version));
```
Comment (jdubois) [inline generators/generator-base.js]: why is this a "test"?

File: generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.YamlPropertiesFactoryBean;
+import org.springframework.boot.SpringApplication;
+import org.springframework.core.io.ClassPathResource;
+
+import java.util.Collections;
+import java.util.Properties;
+
+/**
+ * Utility class to load a spring profile to be used as default
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java]: spring -> Spring

File: generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    private static final long serialVersionUID = 1L;
+
+    private static final Logger log = LoggerFactory.getLogger(DefaultProfileUtil.class);
+
+    private static final String SPRING_PROFILE_ACTIVE = "spring.profiles.active";
+
+    private static final Properties BUILD_PROPERTIES = readProperties();
+
+    /**
+     * get a default profile from application.yml
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java]: get -> Get

File: generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                    activeProfile.contains(Constants.SPRING_PROFILE_PRODUCTION))) {
+                return activeProfile;
+            }
+        }
+        log.warn("No Spring profile configured, running with default profile: {}", Constants.SPRING_PROFILE_DEVELOPMENT);
+        return Constants.SPRING_PROFILE_DEVELOPMENT;
+    }
+
+    /**
+     * set a default to use when no profile is configured.
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java]: set -> Set

File: generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        /*
+        * The default profile to use when no other profiles are defined
+        * This cannot be set in the `application.yml` file.
+        * See https://github.com/spring-projects/spring-boot/issues/1219
+        */
+        app.setDefaultProperties(Collections.singletonMap(SPRING_PROFILE_ACTIVE, getDefaultActiveProfiles()));
+    }
+
+    /**
+     * load application.yml from classpath
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java]: load -> Load , and you need a "." at the end or it's not valid javadoc

File: generators/generator-base.js (hunk 1 above)
Comment (deepu105) [inline generators/generator-base.js]: Oops, my bad, I forgot to change that after debugging. Ill correct it asap

File: generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java (hunk 3 above)
Comment (deepu105) [inline generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java]: Ok

File: generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java (hunk 5 above)
Comment (deepu105) [inline generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java]: Ok missed it here :D

**Comment by deepu105** [comment]
updated all comments

**Comment by erikkemperman** [comment]
@deepu105 I was reading the Spring-Boot guys' response on your issue there, and thought they were suggesting that `spring.profiles.default` ought to work. But you had tried that, obviously, it's in the sample code you gave them. Am I missing something? Where did they say this is a bug on their part?

Just curious. The main thing is it gets resolved somehow, for the moment I guess your workaround is fine.

**Comment by deepu105** [comment]
I meant the original issue with setting active snicoll kind of said thats
its working weird and they would discuss about fixing it so bug or no bug
ill be happy to have it working

**Comment by erikkemperman** [comment]
Agreed! Still, he was pretty sure that `spring.profiles.default` should have worked, right? I was wondering if there's anything specific about JH's DatabaseConfiguration class -- you mentioned seeing this mostly from the wrong database being chosen -- because the profile voodoo there is done with annotations, which I'm not sure during which phase of initialization these are evaluated.

**Comment by deepu105** [comment]
I dont think there is an issue with annotation, which is to load from
external property file. I tried after removing the annotation but same
behaviour

Thanks & Regards,
Deepu

**Comment by erikkemperman** [comment]
All right, it was just a guess...

I just rebased my project on the output from latest JH master, and it all works great!

**Pull request merged**